io/disk/disk_info.py

```diff
@@ -113,24 +115,46 @@ def test(self):
         size, UUID and IO sizes etc
         """
         msg = []
-        disk = (self.disk.split("/dev/"))[1]
-        if process.system("ls /dev/disk/by-id -l| grep -i %s" % disk,
-                          ignore_status=True, shell=True, sudo=True) != 0:
-            msg.append("Given disk %s is not in /dev/disk/by-id" % disk)
-        if process.system("ls /dev/disk/by-path -l| grep -i %s" % disk,
+        disk_nodes = []
+        mpath = False
+        if "mapper" in self.disk:
+            mpath = True
+            disk_abs = os.path.basename(os.readlink(self.disk))
+            disk = (self.disk.split("/dev/mapper/"))[1]
```

Comment:
mpath = self.disk.split("/")[-1]
for better naming convention and better way of split